config/registry.go

```diff
@@ -98,9 +104,29 @@ func Register(option *Option) error {
 		return fmt.Errorf("config: invalid default value: %w", vErr)
 	}
 
+	if err := loadUnmappedValue(option); err != nil {
+		return err
+	}
+
 	optionsLock.Lock()
 	defer optionsLock.Unlock()
 	options[option.Key] = option
 
 	return nil
 }
+
+func loadUnmappedValue(option *Option) error {
+	unmappedValuesLock.Lock()
+	defer unmappedValuesLock.Unlock()
+	if value, ok := unmappedValues[option.Key]; ok {
+		delete(unmappedValues, option.Key)
+
+		var vErr *ValidationError
+		option.activeValue, vErr = validateValue(option, value)
+		if vErr != nil {
+			return fmt.Errorf("config: invalid value: %w", vErr)
+		}
```

Comment:
Normal config loading just collects the validation errors from loaded config and reports them.
This would prevent the plugin from starting with an invalid config value, if it fails when failing to register the config option.

Reply:
after discussion: make sure we can return that error to the caller of `config.Register()` but not fail the registration

Reply:
this requires an update to the plugin system to handle that case correctly